backport.sh

```diff
@@ -98,7 +98,7 @@ cherry_pick() {
   local merge_sha=$4
 
   output=''
-  debug output git clone -q --no-tags -b "${branch}" "${repository}" "${GITHUB_WORKSPACE}" || fail "Unable to clone from repository \`${repository}\` a branch named \`${branch}\`, this should not have happened" &&
+  debug output test -d ${GITHUB_WORKSPACE}/.git && git -C ${GITHUB_WORKSPACE} checkout -b "${branch}" -t "origin/${branch}" || git clone -q --no-tags -b "${branch}" "${repository}" "${GITHUB_WORKSPACE}" || fail "Unable to clone from repository \`${repository}\` a branch named \`${branch}\`, this should not have happened" &&
```

Comment:
```suggestion
  debug output test -d "${GITHUB_WORKSPACE}/.git" && git -C "${GITHUB_WORKSPACE}" checkout -b "${branch}" -t "origin/${branch}" || git clone -q --no-tags -b "${branch}" "${repository}" "${GITHUB_WORKSPACE}" || fail "Unable to clone from repository \`${repository}\` a branch named \`${branch}\`, this should not have happened" &&
```

Reply:
I have added double quotes, and also moved test execution out of debug scope, so that debugging output corresponds to what is described in spec file. This fixed shellcheck and 2 out of 3 shellspec failures. I'll continue looking into 3rd shellspec failure.